### PR TITLE
Add more ix methods

### PR DESCRIPTION
**Pull request by DonDuala**

### CODE REVIEW

**Review by brittcyr — COMMENTED** [review]
add tests

**Comment by DonDuala** [comment]
Updated to a clearOutMarketTxs method since we require the cancelAll to confirm first before running a withdrawAll
@brittcyr give it another sanity check pls

File: client/ts/src/client.ts [hunk 1]
```diff
@@ -361,6 +399,45 @@ export class ManifestClient {
     );
   }
 
+  /**
+   * PlaceOrderWithRequiredDeposit instruction. Only deposits the appropriate base
+   * or quote tokens if not in the withdrawable balances.
+   *
+   * @param payer PublicKey of the trader
+   * @param params PlaceOrderParamsExternal including all the information for
```
Comment (brittcyr) [inline client/ts/src/client.ts:407]: Wrapper in the name

File: client/ts/src/client.ts [hunk 2]
```diff
@@ -361,6 +399,45 @@ export class ManifestClient {
...
+    let depositMint = this.market.baseMint();
+    let depositAmount = params.numBaseTokens - currentBalance;
+
+    if (params.isBid) {
+      depositMint = this.market.quoteMint();
+      depositAmount = params.numBaseTokens * params.tokenPrice - currentBalance;
+    }
+
+    if (depositAmount <= 0) {
+      console.log('Enough balance to place order without deposit');
```
Comment (brittcyr) [inline client/ts/src/client.ts:433]: console.debug

File: client/ts/src/client.ts [hunk 3]
```diff
@@ -361,6 +399,45 @@ export class ManifestClient {
...
+    const placeOrderIx = this.placeOrderIx(params);
+
+    const currentBalance = this.market.getWithdrawableBalanceTokens(
+      payer,
+      !params.isBid,
+    );
+    let depositMint = this.market.baseMint();
+    let depositAmount = params.numBaseTokens - currentBalance;
+
+    if (params.isBid) {
```
Comment (brittcyr) [inline client/ts/src/client.ts:427]: let x;
if () {
x =
} else {
x =
}


to be clearer

File: client/ts/src/client.ts [hunk 4]
```diff
@@ -478,6 +555,67 @@ export class ManifestClient {
...
+          cancelAll: true,
+          orders: [],
+          traderIndexHint: null,
+        },
+      },
+    );
+  }
+
+  /**
+   * ClearOutMarketTxs transactions. Pulls all orders
```
Comment (brittcyr) [inline client/ts/src/client.ts:585]: Generally, a function name that ends in
ix or ixs   returns instructions
tx or txs  returns transcations
neither    does the action

so I would name this just
clearOutMarket

File: client/ts/src/client.ts [hunk 5]
```diff
@@ -478,6 +555,67 @@ export class ManifestClient {
...
+    const cancelAllTx = new Transaction();
+    const cancelAllSig = await sendAndConfirmTransaction(
+      this.connection,
+      cancelAllTx.add(cancelAllIx),
+      [payerKeypair],
+      {
+        commitment: 'confirmed',
+      },
+    );
+    await this.market.reload(this.connection);
```
Comment (brittcyr) [inline client/ts/src/client.ts:606]: Leave a TODO to merge this into one TX. Can either do client side or have a feature request for the wrapper to expose this behavior.

File: client/ts/src/market.ts [hunk 6]
```diff
@@ -201,9 +177,11 @@ export class Market {
       return 0;
     }
     const seat: ClaimedSeat = filteredSeats[0];
-    return isBase
+    const withdrawableBalance = isBase
       ? toNum(seat.baseBalance) / 10 ** this.baseDecimals()
       : toNum(seat.quoteBalance) / 10 ** this.quoteDecimals();
+    console.log('Withdrawable Balance', isBase, withdrawableBalance);
```
Comment (brittcyr) [inline client/ts/src/market.ts:183]: console.debug

**Review by brittcyr — APPROVED** [review]
Make sure to squash when merging

**Pull request merged**